client/src/components/mixins/nr-affiliation-mixin.ts

```diff
@@ -0,0 +1,63 @@
+import Vue from 'vue'
+import { Component } from 'vue-property-decorator'
+import AuthServices from '@/services/auth.services'
+import BusinessServices from '@/services/business.services'
+import newReqModule from '@/store/new-request-module'
+import { BusinessRequest, CreateNRAffiliationRequestBody } from '@/models/business'
+import { NameRequestI } from '@/models'
+
+@Component
+export default class NrAffiliationMixin extends Vue {
+  /**
+   * Create an affiliation between current authenticated Account and current Nr
+   * @param nr The Nr to affiliate
+   */
+  async createAffiliation (nr: NameRequestI): Promise<any> {
+    try {
+      const currentOrganizationId = JSON.parse(sessionStorage.getItem('CURRENT_ACCOUNT')).id
+      const requestBody: CreateNRAffiliationRequestBody = {
+        businessIdentifier: nr.nrNum,
+        phone: nr.applicants?.phoneNumber || '',
+        email: nr.applicants?.emailAddress || ''
+      }
+
+      // Request to affiliate NR to current account
+      const nrResponse = await AuthServices.createNRAffiliation(currentOrganizationId, requestBody)
+      if (nrResponse?.status === 201) {
+      // update the legal api if the status is success
+        const filingBody: BusinessRequest = {
+          filing: {
+            header: {
+              name: 'incorporationApplication',
+              accountId: currentOrganizationId
+            },
+            business: {
+              legalType: nr.entity_type_cd
+            },
+            incorporationApplication: {
+              nameRequest: {
+                legalType: nr.entity_type_cd,
+                nrNumber: nr.nrNum
+              }
+            }
+          }
+        }
+        const filingResponse = await BusinessServices.createBusiness(filingBody)
+        // navigate to manage business dashboard
+        if (filingResponse?.status === 201) {
```

Comment:
Do you need an else block (with a corresponding `throw` so execution goes to the catch block below)?

Same for line 26 above.

Reply:
Good catch, yes! I can't really rely on a 'Created' response can i..

Reply:
Because this code is a try-catch block, if `createBusiness()` fails (and since you have no `.catch()` it will throw an exception), it will be caught.

I'm just thinking there might be other valid response statuses. 2xx 3xx?

Reply:
I'm not sure, i'll have to look in auth. But when they create affiliations locally, the only (successful) response that is watch for was a 'Created' (201) response. I will check again though!

Reply:
Looks like checking for 201 is a "safety check" then. I'm in favour! But what would you do if it wasn't 201?

Reply:
Ah, i knew i missed something. Looking into this quickly :)

Reply:
Your change looks good. If the "if" fails then you throw an error, which is caught by the catch block etc. Great!